tools/gen_selectedops.py

```diff
@@ -1,80 +1,99 @@
-import os
-import sys
+#!/usr/bin/env python3
+
+import argparse
+import pathlib
 
 OPMAP_TO_CMAKE_FLAGS = {
-    'BlingFireSentenceBreaker': 'OCOS_ENABLE_BLINGFIRE',
-    'GPT2Tokenizer': 'OCOS_ENABLE_GPT2_TOKENIZER',
-    'WordpieceTokenizer': 'OCOS_ENABLE_WORDPIECE_TOKENIZER',
-    'StringConcat': 'OCOS_ENABLE_TF_STRING',
-    'StringECMARegexReplace': 'OCOS_ENABLE_TF_STRING',
-    'StringECMARegexSplitWithOffsets': 'OCOS_ENABLE_TF_STRING',
-    'StringEqual': 'OCOS_ENABLE_TF_STRING',
-    'StringToHashBucket': 'OCOS_ENABLE_TF_STRING',
-    'StringToHashBucketFast': 'OCOS_ENABLE_TF_STRING',
-    'StringJoin': 'OCOS_ENABLE_TF_STRING',
-    'StringLength': 'OCOS_ENABLE_TF_STRING',
-    'StringLower': 'OCOS_ENABLE_TF_STRING',
-    'StringRegexReplace': 'OCOS_ENABLE_RE2_REGEX',
-    'StringRegexSplitWithOffsets': 'OCOS_ENABLE_RE2_REGEX',
-    'StringSplit': 'OCOS_ENABLE_TF_STRING',
-    'StringToVector': 'OCOS_ENABLE_TF_STRING',
-    'StringUpper': 'OCOS_ENABLE_TF_STRING',
-    'SegmentExtraction': 'OCOS_ENABLE_MATH',
-    'StringMapping': 'OCOS_ENABLE_TF_STRING',
-    'VectorToString': 'OCOS_ENABLE_TF_STRING',
-    'MaskedFill': 'OCOS_ENABLE_TF_STRING',
-    'BertTokenizer': 'OCOS_ENABLE_BERT_TOKENIZER',
-    'BasicTokenizer': 'OCOS_ENABLE_BERT_TOKENIZER',
-    'BertTokenizerDecoder': 'OCOS_ENABLE_BERT_TOKENIZER',
-    'SentencepieceTokenizer': 'OCOS_ENABLE_SPM_TOKENIZER',
-    'ImageDecode': 'OCOS_ENABLE_VISION',
-    'ImageEncode': 'OCOS_ENABLE_VISION',
+    "BlingFireSentenceBreaker": "OCOS_ENABLE_BLINGFIRE",
```

Comment:
note: ran black for formatting and it made quite a few cosmetic changes here

Reply:
black is pretty controversial for me since it sacrifices some flexibility, and its goal is also questionable, check this article
https://luminousmen.com/post/my-unpopular-opinion-about-black-code-formatter

So, currently it wasn't enabled for this project by default, and was optional.

Reply:
sure, I'm happy to go with whatever formatting convention we want to use for this project. but I didn't know of any so just copied ORT's. let me know if there is another one to follow.

Reply:
Personally I was happy with flake8 enforcing most of PEP8 in ORT. There's a config for that in the ORT repo.

ORT started using black a few months ago, but I wasn't involved in the discussion (assuming there was one) about adopting that as the standard. I also don't like some of the black formatting as I think it diminishes readability in many cases (although #fmt off/#fmt on can turn it off for sections).